server/src/main/java/org/opensearch/index/engine/exec/format/DataFormat.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.engine.exec.format;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.mapper.ParametrizedFieldMapper;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a data format.
+ */
+@ExperimentalApi
+public interface DataFormat {
```

Comment:
Keeping this in core forms a circular dependency in lib - common lib which contains data format codec, but if we move this to lib, then we don't have access to any server classes such as settings.

Reply:
Maybe we need to keep dataFormatCodec in core as well ?